### ISSUE TITLE
Incorrect Rex Color extraction

### ISSUE THREAD

**Issue by SSIDx** [cadon/ARKStatsExtractor#876]
Hi there,

I've seem to come across a bug.
There are two issues, not sure if they are related.

Firstly, the rex color regions are incorrect (according to wiki) rex only has 4 regions, however extraction shows 5. (region 3 isn't used by the Rex and region 4 tool tip isn't correct)

Secondly, Color description isn't catered for? see screen shot below, I have zipped the .ini file if you would like to see the values for the RGBa color conversion.
![image](https://user-images.githubusercontent.com/3681472/51702057-7cb5b600-201b-11e9-8770-87886ff757a5.png)

[DinoExport_50873190366873932.zip](https://github.com/cadon/ARKStatsExtractor/files/2793166/DinoExport_50873190366873932.zip)

Keep up the good work

**Comment by kohonac** [comment]
a quick search here https://ark.gamepedia.com/Rex#Appearance shows that there was an update to the game that wasnt updated in ASB

> Patch 278.0 removed PaintRegion 3 (former Back), Patch 278.2 added PaintRegion 4 as new Back

**Comment by cadon** [comment]
Hi, just closed this issue and now noticed that I'm not quite sure about your second issue.

> Secondly, Color description isn't catered for? see screen shot below, I have zipped the .ini file if you would like to see the values for the RGBa color conversion.

Are the color-names not correct?

**Comment by SSIDx** [comment]
Hi, Sorry for the late response.
The color in the "panel" is displaying correctly, however on the tooltip it says unknown unknown, I have no idea what causes that, but some dinos say unknown DinoDarkBlue for example. (trying to get all colors in all regions, so a near black and black are two entities, and cant tell with naked eye, so was depending on the tooltip)

If it is working on your side then its all good, as im playing on a private unoffical server where there are 14 mods XD cant tell what is installed.

I've also noticed the ark export sometimes export a weird decimal value, on screen the color is black but the rgb isn't the black on the wiki. here is a ini file for the value that is "black" but extractor's closest match is DarkGreen. The other regions are ark's black (RBG at 0.04 each) but some reason the back is at RGB 0.01 [Weird Black.zip](https://github.com/cadon/ARKStatsExtractor/files/2825290/Weird.Black.zip) 

And here is another .ini file for the region4 being unknown unknown [Unknown Unknown.zip](https://github.com/cadon/ARKStatsExtractor/files/2825294/Unknown.Unknown.zip)

Sorry for the lengthy reply. Thought I'd add as much info as I can.

Thanks again for your work.

**Comment by cadon** [comment]
The tooltip contains the name and number of the color-region (first row) and the name and number of the color (second row). If no color is applied, the second row displays `unknown (0)`. If the color-region has no name, it also displays unknown. Regarding the rex, the game-data still has the data for the region3, but it's not used ingame anymore. I'll remove that region manually in ASB, so it won't confuse.

Regarding the unknown colors, it's probably due to a buff like the [Candy](https://ark.gamepedia.com/Festive_Dino_Candy). ARK only knows [56 colors](https://ark.gamepedia.com/Color_IDs), and ASB converts the given RGB-values to the ARK color which is the closest. ARK (or rather the unreal engine) uses not the common RGB-values so they need to be converted first.

Maybe it's related to this [discussion about a strange color](https://www.reddit.com/r/ARK/comments/a6lv3u/how_is_this_color_possible/), interestingly the post is from december 16, when the candies were released only two days later. I think it's still unclear how that color appeared. Maybe it's from a beta-tester who had access to the candies early.

In conclusion, the color-region 3 will be disappear in a next release in ASB and the strange colors will remain strange for now, until it's clear what extra colorization mechanic ARK uses (are these colors permament, will they be inherited to offspring?).

**Comment by SSIDx** [comment]
Yeah, I understand the swap from region 3 to 4. I just didn't understand the weird colors that weren't in the known 56 colors and yeah, the colors are permanent, I didnt tame things for the colors and the rex was before event and is a 3rd or 4th generation, cant remember. Well there are some stuff that just cant be solved, maybe a mod is messing with the ark export because at times ASB is able to determine the color for that zone.

Thanks again

**Comment by SSIDx** [comment]
The color is definitely being passed on, the mutation tree is 1/-174600161 
well more accurately 8 pairs from maternal and 15 pairs on paternal. 

The baseline for the .ini rex is 0/0, 6 pairs maternal and 6 pairs paternal. and its the same color. So its safe to say that its being passed from generation to generation.

Hmm I don't know if i should of closed the issue or left it for you to do?

**Comment by cadon** [comment]
No problem, the one issue was resolved indeed. I'll say we leave it open for the second issue with the extra colors.

**Comment by SSIDx** [comment]
Cool, will re open it, thanks for the timely responses, Thanks again

**Comment by cadon** [comment]
All colors should be supported by now, so I'll close this issue.
